tests/test-stake-pool-sdk.ts

```diff
@@ -178,6 +180,81 @@ describe("test basic functionalities", () => {
       );
     });
 
+    it("it calcSolDeposit() and calcSolDepositsInverse() works correctly", async () => {
+      const socean = new Socean("testnet");
+      const stakePool = await socean.getStakePoolAccount();
+
+      const depositAmountSol = Math.random() / 4;
+      const depositAmount = Math.round(depositAmountSol * LAMPORTS_PER_SOL);
+      const depositAmountLamports = new Numberu64(depositAmount);
+      const { dropletsReceived, dropletsFeePaid, referralFeePaid } =
+        calcSolDeposit(depositAmountLamports, stakePool.account.data);
+
+      const {
+        lamportsStaked,
+        dropletsFeePaid: inverseDropletsFeePaid,
+        referralFeePaid: inverseReferralFeePaid,
+      } = calcSolDepositInverse(dropletsReceived, stakePool.account.data);
+
+      expect(
+        depositAmountLamports.toNumber() - lamportsStaked.toNumber(),
+      ).to.be.at.most(2);
+      expect(dropletsFeePaid.toNumber()).to.eq(
+        inverseDropletsFeePaid.toNumber(),
+      );
+      expect(referralFeePaid.toNumber()).to.eq(
+        inverseReferralFeePaid.toNumber(),
+      );
+    });
+
+    it("it calcWithdrawals() and calcWithdrawalsInverse() works correctly", async () => {
+      const socean = new Socean("testnet");
+      const stakePool = await socean.getStakePoolAccount();
+      const validatorList = await socean.getValidatorListAccount(
+        stakePool.account.data.validatorList,
+      );
+      const scnSolAtaAcctInfo = await scnSolToken.getAccountInfo(
+        scnSolAtaPubkey,
+      );
+      const initialBalanceDroplets = scnSolAtaAcctInfo.amount;
+
+      const withdrawAmountDroplets = new Numberu64(
+        Math.round(Math.random() * initialBalanceDroplets.toNumber()),
+      );
```

Comment:
you can probably just use random numbers lower than total stake pool token supply here since we're just checking the calculation results and not sending any actual transactions